### PR TITLE
Added editing functions

### PR DESCRIPTION
**Pull request by haefnir**
No deletion functionality yet, will be next task

### CODE REVIEW

File: edit.php [hunk 1]
```diff
@@ -14,10 +14,10 @@
     <title>Edit Pokemon</title>
 </head>
 <body>
-<h1>Edit Pokemon</h1>
+<h1>Edit <?=($userPokemon['hasNickname']==1) ? $userPokemon['nickname'] . " the " : ""; echo $userPokemon['species'];?></h1>
 <?php echo displayPokemon($userPokemon,0); ?>
-<form>
-    <label>Give <?php echo $userPokemon['nickname'] ?? $userPokemon['species'];?> a new nickname?
+<form method="post" action="editNickname.php">
+    <label>Give <?php echo $userPokemon['nickname'] ?? $userPokemon['species'];?> a new nickname? (Leave blank to clear nickname. Max chars: 12)
```
Comment (doubt-ur-vibe) [inline edit.php:20]: Maybe you could have a remove nickname button instead?

File: editNickname.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+require_once "functions/allFunctions.php";
+
+if (!isset($_POST['id'])){
+    header("location: index.php");
+    exit();
+}
+if (!is_numeric($_POST['id'])){
+    header("location: index.php");
+    exit();
```
Comment (doubt-ur-vibe) [inline editNickname.php:10]: Maybe you could move both of these if statements into one with an or operator (||)

File: editNickname.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+$db = connectToDB('pokemon-collection');
+$inputtedNewNickname = $_POST['newNickname'];
+$inputtedID = $_POST['id'];
+
+$cleanNickname = sanitiseText($inputtedNewNickname);
+$cleanID = sanitiseText($inputtedID);
+
+changeNicknameInDatabase($cleanID, $cleanNickname, $db);
+
+header("location:edit.php?id={$_POST['id']}");
```
Comment (doubt-ur-vibe) [inline editNickname.php:22]: Maybe you could turn some of the code in this file into functions

**Review by doubt-ur-vibe — APPROVED** [review]
Looking good broski :)

File: editNickname.php (hunk 2 above)
Comment (haefnir) [reply editNickname.php:10]: Combined them into an AND statement as this logic table is the the same as ! OR !

File: edit.php (hunk 1 above)
Comment (haefnir) [reply edit.php:20]: Have considered it, but prefer this method for now. Might add later though!

**Pull request merged**